### PR TITLE
[HZ-762] Add THIRD-PARTY.txt and attribution.txt to Hazelcast artifacts [4.2.z]

### PR DESCRIPTION
**Pull request by kwart**
This PR adds the `THIRD-PARTY` and `attribution` files into `4.2.z` branch.

Backport of: #19927, #18573

### CODE REVIEW

**Review by olukas — APPROVED** [review]
LGTM. The only minor thing is that we also include couple of HZ projects in attribution:
```
com.hazelcast:hazelcast-aws:3.3
	Copyright 2020 Hazelcast Inc.

com.hazelcast:hazelcast-azure:2.1
	Copyright 2020 Hazelcast Inc.

com.hazelcast:hazelcast-gcp:2.1
	Copyright 2020 Hazelcast Inc.

com.hazelcast:hazelcast-hibernate53:2.1.1
	Copyright 2020 Hazelcast Inc.

com.hazelcast:hazelcast-kubernetes:2.2.2
	Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.

com.hazelcast:hazelcast-wm:4.0
	Copyright 2020 Hazelcast Inc.
```
However I don't think it's a problem.

File: distribution/pom.xml [hunk 1]
```diff
@@ -27,7 +27,10 @@
 
     <name>hazelcast-distribution</name>
     <artifactId>hazelcast-distribution</artifactId>
-    <packaging>pom</packaging>
```
Comment (frant-hartm) [inline distribution/pom.xml:30]: Won't this cause a jar being generated?

**Review by frant-hartm — APPROVED** [review]
Just one minor question.

File: distribution/pom.xml (hunk 1 above)
Comment (kwart) [reply distribution/pom.xml:30]: Good point. The jar will be generated, but we don't want to deploy it. So I'm adding a new commit with a change in `maven-deploy-plugin` configuration.

**Pull request merged**